http://cygwin.com/ml/cygwin-xfree/2009-06/msg00071.html

It seems that the (new for Cygwin 1.7) getifaddr() function can return 
interfaces with IFF_BROADCAST & IFF_UP set, but no broadcast address, 
which the X server assumes never happens.

--- origsrc/xorg-server-1.6.0/os/access.c	2009-02-25 19:12:13.000000000 +0000
+++ src/xorg-server-1.6.0/os/access.c	2009-06-29 11:22:03.781250000 +0100
@@ -812,7 +812,8 @@ DefineSelf (int fd)
 		continue;
 #endif
 	    if ((ifr->ifa_flags & IFF_BROADCAST) &&
-		(ifr->ifa_flags & IFF_UP))
+		(ifr->ifa_flags & IFF_UP) &&
+                ifr->ifa_broadaddr)
 		broad_addr = *ifr->ifa_broadaddr;
 	    else
 		continue;

